### PR TITLE
support any application with save-ansys-path

### PR DESCRIPTION
**Pull request by koubaa**
Allow saving any path, not just well-known paths using ansys-tools-path. I want to use this for another program that doesn't belong to any installation (it uses a portable install) and has no version number (it is versioned independently from the likes of mapdl, mechanical, etc).

Obviously the features to find the location and get the version of the application would not work, but I just want to use `save-ansys-path` to store it and to get the path in the python client of that other program.

I tested this locally using 7zip, like this:

```
$ save-ansys-path --name=7z "C:\path\to\7-Zip\7zG.exe"
$ python
>>> from ansys.tools.path.path import _get_application_path
>>> _get_application_path("7z")
'C:\\path\\to\\7-Zip\\7zG.exe'
```

As an implementation detail, I added a simple plugin-system for the "supported" applications (dyna, mapdl, mechanical). I didn't migrate much there, but I think we should move app-specific code there for better maintenance going forwards.

### CODE REVIEW

File: src/ansys/tools/path/path.py [hunk 1]
```diff
@@ -9,11 +9,14 @@
 
 import platformdirs
 
+from ansys.tools.path.applications import ApplicationPlugin, dyna, mapdl, mechanical
 from ansys.tools.path.misc import is_float, is_linux, is_windows
 
+PLUGINS: Dict[str, ApplicationPlugin] = {"mechanical": mechanical, "dyna": dyna, "mapdl": mapdl}
```
Comment (germa89) [inline src/ansys/tools/path/path.py:15]: I wonder how we are going to populate this with new applications. You did mention. the example for 7zip, and I guess "7zip" should appear here. Somehow.

I guess we need to use something to store the new apps?? Maybe a txt file in a settings directory? Something like this?

https://github.com/ansys/pymapdl/blob/1a720959ab3e91dce20074b514d4faeb7707e11a/src/ansys/mapdl/core/__init__.py#L32-L37

**Review by germa89 — APPROVED** [review]
Overall, it seems a good approach.

However I am not particularly keen on using modules as classes (using ``dyna.is_valid_executable_path``). It seems you were going to use a base class first, (that's probably you defined ``ApplicationPlugin`` class), but then you just used the modules (python files) as classes. Although correct, I personally don't like it. But that's your call.

File: src/ansys/tools/path/path.py (hunk 1 above)
Comment (koubaa) [reply src/ansys/tools/path/path.py:15]: @germa89 this is just an implementation detail. Ideally there would be a way to install plugins. I would want to learn about the `pluggy` library in python which seems very well suited for adding plugins.

In the case of 7zip or other "unregistered" applications, those would not have plugins. The whole point here is to allow `save` & `get` for anything, but `find` only for registered applications.

Comment (germa89) [reply src/ansys/tools/path/path.py:15]: Ok I see. Fine with me then.

**Pull request merged**